### PR TITLE
Update developer guide, user guide, about us and PPP

### LINKED ISSUE #166: [PE-D][Tester A] Mismatch in expectations for user guide for `findo` command
# Issue

The user guide specifies that `At least one of the 3 prefixes must be used in the findo command` which is not true, since the `findo` command works even without any of the 3 prefixes, such as `findo i/Table`.

# Suggestion

It might be a typo in the user guide, which can be misleading for the users.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: EmilyOng/ped#7

### LINKED ISSUE #198: Draft Skeletal PPP (Janelle)

### PR DESCRIPTION
**Pull request by janelleljt**
In this PR,
- Added a Using this guide and tutorial segment to UG
- Slightly restructured the UG
- Updated DG with sequence diagrams for `findo`
- Updated my own PPP
- Updated AboutUs.md to reflect roles and responsibilities

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -21,27 +21,30 @@ If you are looking to help develop this project, take a look at our [Developer G
...
+* [Features](#features)
+  * [Inventory management](#uinventory-managementu)
     * [Adding an inventory item: `addi`](#adding-an-inventory-item-addi)
-    * [Listing all inventory items: `list`](#listing-all-inventory-items-listi)
+    * [Listing all inventory items: `listi`](#listing-all-inventory-items-listi)
     * [Finding an inventory item: `findi`](#finding-an-inventory-item-findi)
     * [Deleting an inventory item: `deletei`](#deleting-an-inventory-item-deletei)
     * [Editing an inventory item: `editi`](#editing-an-inventory-item-editi)
-  * Order management
+  * [Order management](#uorder-managementu)
```
Comment (donljh) [inline docs/UserGuide.md:36]: Just wanted to clarify -- why does this link have the leading and trailing `u` characters?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+
+As a first time user, we understand that it might be daunting to start up the application without sufficient instructions. So, we have prepared
+a [Quick start](#quick-start) section in this guide to equip you with the knowledge to start up TrackO for your business.
+
+Next, after starting up TrackO, you might be confused by the unfamiliar layout. As such, we have prepared a breakdown of the **layout** which
+you can find [here](#layout). This section of the guide will help you understand the different sections of the [GUI](#gui) as well as 
+their roles in this application. 
+
+After familiarising yourself with the [GUI](#gui), we have included a [tutorial section](#tutorial) where you can use some 
+example commands play around with some of our sample data and get to know TrackO's features a bit better.
```
Comment (donljh) [inline docs/UserGuide.md]: ...example commands to play around...

File: docs/UserGuide.md [hunk 3]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+their roles in this application. 
+
+After familiarising yourself with the [GUI](#gui), we have included a [tutorial section](#tutorial) where you can use some 
+example commands play around with some of our sample data and get to know TrackO's features a bit better. 
+
+Following that, as our application uses commands which require specific inputs. To help you out with learning the command formats used
+in this guide, you can head over to the [Command Guide](#command-guide) section of this guide. This will come in handy for you when 
+you're ready to get some in-depth exposure to our [features](#features). Because our applications helps your business handle 
+orders and inventory, we have segmented our commands into 2 sections. You can click [here](#uinventory-managementu) to find commands 
+related to **managing orders** and click [here](#uorder-managementu) to find commands which help with **inventory management**. For more **general features**,
```
Comment (donljh) [inline docs/UserGuide.md]: I think you swapped the links for order and inventory management

File: docs/UserGuide.md [hunk 4]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+4. Double-click the file to start the app. The GUI similar to the one below should appear in a few seconds. The application will already contain some sample
+   data.<br>
    ![Ui](images/Ui.png)
 
-1. Type the command in the command box and press Enter to execute it. e.g. typing **`listo`** and pressing Enter will show a list of existing orders.<br>
-   Some example commands you can try:
+5. To start out with some commands, head over to the [tutorial segment](#tutorial) of this user guide!
 
-    * **`listo`** : Lists all orders.
+6. If you'd like to skip the tutorial, head over to the [features section](#features) below for some of the
```
Comment (donljh) [inline docs/UserGuide.md]: Might want to standardise segment vs section?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+5. To start out with some commands, head over to the [tutorial segment](#tutorial) of this user guide!
 
-    * **`listo`** : Lists all orders.
+6. If you'd like to skip the tutorial, head over to the [features section](#features) below for some of the 
+more technical specifications of each command.
 
-    * **`addi`**`n/Keychain q/200 d/This is a metal keychain sp/15.00 cp/10.00` : Adds an inventory item named `Keychain`, which has quantity `200`, description `This is a metal keychain`, selling price `$15.00` and cost price `$10.00` to the inventory list.
+<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
+Want to begin integration with TrackO now? Enter the `clear` command to get rid of the sample data.
+</div>
```
Comment (donljh) [inline docs/UserGuide.md]: I think you can change this to

Want to start using TrackO from scratch? Enter the `clear` command to remove all application data.

And maybe link the clear command as well.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+## Layout
 
-    * **`exit`** : Exits the app.
+--------------------------------------------------------------------------------------------------------------------
+## Tutorial
+
+To get you started, here are some simple commands you can try out on our sample data before getting to know more of
+TrackO's functionalities! 
+
+1. We start off by getting to know how to manage our **inventory**!
```
Comment (donljh) [inline docs/UserGuide.md]: ...manage your inventory

File: docs/UserGuide.md [hunk 7]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+   1. Adding an item: `addi`
+       * Say your small business has `350` of a new item, `Teddy bear keychain` currently in your physical inventory.
```
Comment (donljh) [inline docs/UserGuide.md]: ...`350` units of the item `Teddy bear keychain`...

File: docs/UserGuide.md [hunk 8]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+         It cost you `$0.89` to make these keychains and are selling them at `$3.50`.
```
Comment (donljh) [inline docs/UserGuide.md]: It costs you...... and your selling price for each unit is `$3.50`

File: docs/UserGuide.md [hunk 9]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+   You should see this order appear in TrackO at index 5 as shown below:
+
+<div markdown="span" class="alert alert-primary">:bulb: **Note:**
+If an item is not in your inventory, TrackO does not allow you to add that item to an order!
+</div>
 
-1. Refer to the [Features](#features) below for details of each command.
+   2. Finding order(s): `findo`
+      * Our application allows you to search for orders by different fields (eg. payment status, delievery status, customer details).
+      * To find orders to be sent to `Clementi` which have not been paid enter the following command to the [command box](link to layout section):
```
Comment (donljh) [inline docs/UserGuide.md]: ...to be delivered to... that are not paid for, ...

File: docs/UserGuide.md [hunk 10]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+      findo -P a/Clementi
+      ```
+   You should see these orders appear in the [order card](link to layout).
 
---------------------------------------------------------------------------------------------------------------------
+   3. Listing order(s): `listo`
+       * To view all your orders again, enter `listo`.
 
-## Features
+   You should see these orders appear in the [order card](link to layout).
```
Comment (donljh) [inline docs/UserGuide.md]: I think it should be

appear in the order list 

instead of order card

File: docs/UserGuide.md (hunk 4 above)
Comment (donljh) [reply docs/UserGuide.md]: Do check this for other parts of the document too

**Review by donljh — CHANGES_REQUESTED** [review]
Good work! Thanks for the effort for all the sections included in this PR. I've added some minor comments along the way.

File: docs/UserGuide.md [hunk 11]
```diff
@@ -69,59 +72,141 @@ Currently, **TrackO** only supports tracking inventory items that are countable
...
+in this guide, you can head over to the [Command Guide](#command-guide) section of this guide. This will come in handy for you when 
+you're ready to get some in-depth exposure to our [features](#features). Because our applications helps your business handle 
+orders and inventory, we have segmented our commands into 2 sections. You can click [here](#uinventory-managementu) to find commands 
+related to **managing orders** and click [here](#uorder-managementu) to find commands which help with **inventory management**. For more **general features**, 
+you can go to [this section](#ugeneral-featuresu) as well!
+
+Lastly, we have included a handy [command summary](#command-summary) for when you're proficient at using TrackO and just want to take a quick look
+at any commands you need!
+
+p.s. Not too sure about some technical terms in this guide? Check out our [glossary](#glossary) to get better understand all the technical jargon!
```
Comment (sohenze) [inline docs/UserGuide.md]: to get a better understanding of all the...

File: docs/UserGuide.md (hunk 8 above)
Comment (sohenze) [reply docs/UserGuide.md]: Perhaps "costed" ? 
Or "costs" you...per unit

**Review by sohenze — APPROVED** [review]
Generally looks good after the changes mentioned are made!

File: docs/UserGuide.md (hunk 8 above)
Comment (melissaharijanto) [reply docs/UserGuide.md]: I think 'costs' is better!

File: docs/UserGuide.md (hunk 1 above)
Comment (janelleljt) [reply docs/UserGuide.md:36]: The trailing u characters were to underline those headers! just to distinguish them a bit from the actual commands

**Review by donljh — APPROVED** [review]

**Pull request merged**